src/vmod_vsthrottle.c

```diff
@@ -303,6 +303,35 @@ vmod_blocked(VRT_CTX, VCL_STRING key, VCL_INT limit, VCL_DURATION period,
 	return (ret);
 }
 
+VCL_VOID
+vmod_remove_bucket(VRT_CTX, VCL_STRING key, VCL_INT limit, VCL_DURATION period,
+		   VCL_DURATION block)
+{
+	struct tbucket *b;
+	struct vsthrottle *v;
+	struct tbucket k;
+	unsigned char digest[SHA256_LEN];
+	unsigned part;
+
+	(void)ctx;
+
+    do_digest(digest, key, limit, period, block);
```

Comment:
you should be able to use to either use `get_bucket`, or at the very least point `do_digest` at `&k.digest` and save yourself a `memcpy`, no?